File: c#/ConcreteStatService.cs
using System.Collections.Generic;
using System.Threading.Tasks;

namespace Backend;

public class ConcreteStatService : IStatService
{
    public IDictionary<string, int> GetCountryPopulations()
    {
        // Pretend this calls a REST API somewhere
		return new Dictionary<string, int>
		{
			{ "India", 1182105000 },
			{ "United Kingdom", 62026962 },
			{ "Chile", 17094270 },
			{ "Mali", 15370000 },
			{ "Greece", 11305118 },
			{ "Armenia", 3249482 },
			{ "Slovenia", 2046976 },
			{ "Saint Vincent and the Grenadines", 109284 },
			{ "Bhutan", 695822 },
			{ "Aruba (Netherlands)", 101484 },
			{ "Maldives", 319738 },
			{ "Mayotte (France)", 202000 },
			{ "Vietnam", 86932500 },
			{ "Germany", 81802257 },
			{ "Botswana", 2029307 },
			{ "Togo", 6191155 },
			{ "Luxembourg", 502066 },
			{ "U.S. Virgin Islands (US)", 106267 },
			{ "Belarus", 9480178 },
			{ "Myanmar", 59780000 },
			{ "Mauritania", 3217383 },
			{ "Malaysia", 28334135 },
			{ "Dominican Republic", 9884371 },
			{ "New Caledonia (France)", 248000 },
			{ "Slovakia", 5424925 },
			{ "Kyrgyzstan", 5418300 },
			{ "Lithuania", 3329039 },
			{ "United States of America", 309349689 }
		};
    }

    public async Task<IDictionary<string, int>> GetCountryPopulationsAsync()
    {
        return await Task.FromResult(GetCountryPopulations());
    }
}

File: backend/src/main/java/com/quickbase/devint/ConcreteStatService.java
package com.quickbase.devint;

import java.util.ArrayList;
import java.util.List;

import lombok.extern.slf4j.Slf4j;
import org.apache.commons.lang3.tuple.ImmutablePair;
import org.apache.commons.lang3.tuple.Pair;
import reactor.core.publisher.Mono;

@Slf4j
public class ConcreteStatService implements IStatService {

	/**
	 * Returns an unordered list of countries and their populations
	 */
	@Override
	public Mono<List<Pair<String, Integer>>> getCountryPopulations() {
		final var output = new ArrayList<Pair<String, Integer>>();
		
		// Pretend this calls a REST API somewhere
		output.add(new ImmutablePair<>("India", 1182105000));
		output.add(new ImmutablePair<>("United Kingdom",62026962));
		output.add(new ImmutablePair<>("Chile",17094270));
		output.add(new ImmutablePair<>("Mali",15370000));
		output.add(new ImmutablePair<>("Greece",11305118));
		output.add(new ImmutablePair<>("Armenia",3249482));
		output.add(new ImmutablePair<>("Slovenia",2046976));
		output.add(new ImmutablePair<>("Saint Vincent and the Grenadines",109284));
		output.add(new ImmutablePair<>("Bhutan",695822));
		output.add(new ImmutablePair<>("Aruba (Netherlands)",101484));
		output.add(new ImmutablePair<>("Maldives",319738));
		output.add(new ImmutablePair<>("Mayotte (France)",202000));
		output.add(new ImmutablePair<>("Vietnam",86932500));
		output.add(new ImmutablePair<>("Germany",81802257));
		output.add(new ImmutablePair<>("Botswana",2029307));
		output.add(new ImmutablePair<>("Togo",6191155));
		output.add(new ImmutablePair<>("Luxembourg",502066));
		output.add(new ImmutablePair<>("U.S. Virgin Islands (US)",106267));
		output.add(new ImmutablePair<>("Belarus",9480178));
		output.add(new ImmutablePair<>("Myanmar",59780000));
		output.add(new ImmutablePair<>("Mauritania",3217383));
		output.add(new ImmutablePair<>("Malaysia",28334135));
		output.add(new ImmutablePair<>("Dominican Republic",9884371));
		output.add(new ImmutablePair<>("New Caledonia (France)",248000));
		output.add(new ImmutablePair<>("Slovakia",5424925));
		output.add(new ImmutablePair<>("Kyrgyzstan",5418300));
		output.add(new ImmutablePair<>("Lithuania",3329039));
		output.add(new ImmutablePair<>("United States of America",309349689));

		return Mono.just(output);
	}

}
